scipio/src/executor.rs

```diff
@@ -1193,3 +1235,39 @@ fn test_detach() {
         Timer::new(Duration::from_micros(100)).await;
     });
 }
+
+#[test]
+fn test_spin() {
+    use core::mem::MaybeUninit;
+
+    let mut s0 = MaybeUninit::<libc::rusage>::uninit();
+    let mut e0 = MaybeUninit::<libc::rusage>::uninit();
+
+    fn getrusage(ptr: *mut libc::rusage) {
+        let err = unsafe { libc::getrusage(libc::RUSAGE_SELF, ptr) };
+        if err != 0 {
+            panic!("getrusage error = {}", err);
+        }
+    }
+
+    let local_ex =
+        LocalExecutor::with_spin(Duration::from_secs(5)).expect("failed to create local executor");
+
+    let now = Instant::now();
+
+    getrusage(s0.as_mut_ptr());
+    local_ex.run(async {});
+    getrusage(e0.as_mut_ptr());
+
+    let start = unsafe { s0.assume_init() };
+    let end = unsafe { e0.assume_init() };
+    println!("elapsed time = {}", now.elapsed().as_micros());
```

Comment:
those prints would all be ignored and not even seen unless we pass `--nocapture`.

That is the wrong way to test this. The right way is to make sure that by looking at `rusage` we are seeing a lot more time alloted to our thread than what we thought we used.

To make sure the test is not flaky you should run it once with the system free, and another time pinned to a cpu that has many other tasks running (like just `yes`). Both versions should pass.

Reply:
The code is in the middle of the development process. PR is a draft. I was half-way thro. Don't look at it until it's not a draft.